[LP-50733; LP-50734; LP-50735] Feat: Implements creation of folders - mkdir

l-mansouri opened the pull request:
## Overview

This PR does the following:

- Adds command to create folders
`cloudos datasets mkdir <parent_path> <new_folder_name> `

## Jira

[LP-50733](https://lifebit.atlassian.net/browse/LP-50733)
[LP-50734](https://lifebit.atlassian.net/browse/LP-50734)
[LP-50735](https://lifebit.atlassian.net/browse/LP-50735)

# IMPLEMENTATION HAS BEEN CHANGED, PLEASE LOOK AT THE COMMENTS!!

## `cloudos datasets mkdir --help`

![Screenshot 2025-06-25 at 15 50 01](https://github.com/user-attachments/assets/d81639e5-014b-4ff7-9fd9-6e0695741290)

## Tests

First, configure your profile using `cloudos configure --profile <name>`. 


> **Note:** The following examples are run in internal prod cloudos, with project `leila-testing`. 

### Scenario 1 - Successfully create a virtual folder in a project dataset

#### Creating a folder at the dataset level
`cloudos datasets mkdir Data my_new_folder`
![Screenshot 2025-06-25 at 15 55 22](https://github.com/user-attachments/assets/28790854-2f90-4c23-a23b-c63d994d3c13)
![Screenshot 2025-06-25 at 15 56 54](https://github.com/user-attachments/assets/faea7841-bfac-432a-bbd6-73459a630aee)
#### Creating a folder in a subfolder
`cloudos datasets mkdir Data/test new_folder`
![Screenshot 2025-06-25 at 16 00 25](https://github.com/user-attachments/assets/f1bd75bd-06c6-4554-ba73-c442e15ac4b0)

### Scenario2 - Prevent folder creation in restricted areas
![Screenshot 2025-06-25 at 16 01 38](https://github.com/user-attachments/assets/98ad1eff-af7c-4ca3-844a-3def36483876)



[LP-50733]: https://lifebit.atlassian.net/browse/LP-50733?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LP-50734]: https://lifebit.atlassian.net/browse/LP-50734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LP-50735]: https://lifebit.atlassian.net/browse/LP-50735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dapineyro reviewed (changes requested):
@l-mansouri apart from my comments, I have a general comment about the implementation: why did you implement
```
cloudos datasets mkdir <parent_path> <new_folder_name>
```
It is a bit counterintuitive. I'd rather implement:
```
cloudos datasets mkdir Data/<subfolder_if_any>/<new_folder_name>
```
Or in other words, just one input path, like the normal mkdir.

l-mansouri commented:
@dapineyro I have addressed the comments and changed the implementation, here are the scenarios tested with the new implementation

## `cloudos datasets mkdir --help`
![Screenshot 2025-07-01 at 11 37 27](https://github.com/user-attachments/assets/038bfeb8-23d6-44d0-82e4-892a4790e1d6)

## Tests

First, configure your profile using `cloudos configure --profile <name>`. 


> **Note:** The following examples are run in internal prod cloudos, with project `leila-testing`. 

### Scenario 1 - Successfully create a virtual folder in a project dataset

#### Creating a folder at the dataset level
`cloudos datasets mkdirData/my_folder`
![Screenshot 2025-07-01 at 11 31 09](https://github.com/user-attachments/assets/0768aacf-c662-4c01-ac93-df3204854442)

#### Creating a folder in a subfolder
`cloudos datasets mkdir Data/my_folder/test_folder`
![Screenshot 2025-07-01 at 11 34 25](https://github.com/user-attachments/assets/14369d19-e538-4092-b90e-363405d3f1b9)


### Scenario2 - Prevent folder creation in restricted areas
![Screenshot 2025-07-01 at 11 39 22](https://github.com/user-attachments/assets/c0b1534f-0e1d-4fc6-ade9-45be9a1e2d20)
